datasets/turkish_product_reviews/turkish_product_reviews.py

```diff
@@ -0,0 +1,64 @@
+"""Turkish Product Reviews"""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import os
+
+import datasets
+
+
+_CITATION = ""
+
+_DESCRIPTION = """
+Turkish Product Reviews.
+This repository contains 235.165 product reviews collected online. There are 220.284 positive, 14881 negative reviews.
+"""
+
+_URL = "https://github.com/fthbrmnby/turkish-text-data/raw/master/reviews.tar.gz"
+_FILES_PATHS = ["reviews.pos", "reviews.neg"]
+
+_HOMEPAGE = "https://github.com/fthbrmnby/turkish-text-data"
+
+
+class TurkishProductReviews(datasets.GeneratorBasedBuilder):
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "sentence": datasets.Value("string"),
+                    "sentiment": datasets.ClassLabel(names=["positive", "negative"]),
+                }
+            ),
+            citation=_CITATION,
+            homepage=_HOMEPAGE,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        dl_paths = dl_manager.download_and_extract(_URL)
+        filenames = list(os.listdir(dl_paths))
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN, gen_kwargs={"filepath": dl_paths, "filenames": filenames}
+            ),
+        ]
+
+    def _generate_examples(self, filepath, filenames):
+        """Generate TurkishProductReviews examples."""
+        logging.info("⏳ Generating examples from = %s", filepath)
+        for f in filenames:
```

Comment:
```suggestion
        for f in sorted(filenames):
```
we want the order of the files to be deterministic